src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,137 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = head.prev;
+            size++;
+            return;
+        }
+
+        Node<T> iteratorNode = iterator(index);
+        Node<T> newNode = new Node<>(iteratorNode.prev, value, iteratorNode);
+        iteratorNode.prev.next = newNode;
+        iteratorNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return iterator(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T currentValue = iterator(index).value;
+        iterator(index).value = value;
+        return currentValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T currentValue = iterator(index).value;
+        unlink(iterator(index));
+        return currentValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+
+        while (currentNode != null) {
+            if ((currentNode.value == object) || currentNode.value != null
+                    && currentNode.value.equals(object)) {
```

Comment:
this brackets are redundant (read about [Operator Precedence](https://www.programiz.com/java-programming/operator-precedence))